Mathlib/RingTheory/Spectrum/Prime/Topology.lean

```diff
@@ -422,6 +428,32 @@ lemma primeSpectrumProd_symm_inr (x) :
     (primeSpectrumProd R S).symm (.inr x) = comap (RingHom.snd R S) x := by
   ext; simp [Ideal.prod]
 
+lemma range_comap_fst :
+    Set.range (comap (RingHom.fst R S)) = zeroLocus (RingHom.ker (RingHom.fst R S)) := by
+  refine Set.ext fun p ↦ ⟨?_, fun h ↦ ?_⟩
+  · rintro ⟨I, hI, rfl⟩; exact Ideal.comap_mono bot_le
+  obtain ⟨p, hp, eq⟩ | ⟨p, hp, eq⟩ := p.1.ideal_prod_prime.mp p.2
+  · exact ⟨⟨p, hp⟩, PrimeSpectrum.ext <| by simpa [Ideal.prod] using eq.symm⟩
+  · refine (hp.ne_top <| (Ideal.eq_top_iff_one _).mpr ?_).elim
+    simpa [eq] using h (show (0, 1) ∈ RingHom.ker (RingHom.fst R S) from rfl)
```

Comment:
Can you replace the `rfl` in `from rfl` by something else?

Similarly below.

Reply:
You could do `RingHom.mem_ker.mpr rfl` but I don't see the point ... and mathlib doesn't have a lemma for `(x,y).1=x` apparently.

Reply:
I guess `by simp` does it (also `simp only` without any lemma). I usually try to avoid `rfl` since it depends on the actual definition rather than the API, but I agree it's extremely unlikely that `(a,b).1=a` will not be a definitional equality. So don't worry.